confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeConfCheck.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.confignode.conf;
+
+import org.apache.iotdb.confignode.exception.conf.RepeatConfigurationException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.Properties;
+
+/**
+ * ConfigNodeConfCheck checks parameters in iotdb-confignode.properties when started, and the
+ * consistency of some specific parameters on restart
+ */
+public class ConfigNodeConfCheck {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ConfigNodeConfCheck.class);
+
+  private static final ConfigNodeConf conf = ConfigNodeDescriptor.getInstance().getConf();
+
+  private Properties specialProperties;
+
+  public void checkConfig() throws RepeatConfigurationException, IOException {
+
+    String propsDir = ConfigNodeDescriptor.getInstance().getPropsDir();
+    if (propsDir == null) {
+      // Skip configuration check when developer mode or test mode
+      return;
```

Comment:
Read the configuration and check whether the configuration can be shared with the standalone version.